src/xcode/ENA/ENA/Source/Workers/AppUpdateCheckHelper.swift

```diff
@@ -0,0 +1,108 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import Foundation
+import UIKit
+
+enum UpdateAlertType {
+	case none
+	case update
+	case forceUpdate
+}
+
+class AppUpdateCheckHelper {
+
+	let client: Client
+
+    /// The retained `NotificationCenter` observer that listens for `UIApplication.didBecomeActiveNotification` notifications.
+    var applicationDidBecomeActiveObserver: NSObjectProtocol?
+
+	init(client: Client) {
+		self.client = client
+	}
+	deinit {
+		removeObserver()
+	}
+
+	func checkAppVersionDialog(for vc: UIViewController?) {
+		client.appConfiguration { result in
+			guard let versionInfo: SAP_ApplicationVersionConfiguration = result?.appVersion else {
+				return
+			}
+			guard let appVersion: String = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String else {
+				return
+			}
+			let minVersion = "\(versionInfo.ios.min.major).\(versionInfo.ios.min.minor).\(versionInfo.ios.min.patch)"
+			let latestVersion = "\(versionInfo.ios.latest.major).\(versionInfo.ios.latest.minor).\(versionInfo.ios.latest.patch)"
+			guard let alert = self.createAlert(self.compareVersion(currentVersion: appVersion, minVersion: minVersion, latestVersion: latestVersion), vc: vc) else {
+				return
+			}
+			vc?.present(alert, animated: true, completion: nil)
+		}
+	}
+
+	private func setObserver(vc: UIViewController?) {
+		guard self.applicationDidBecomeActiveObserver == nil else { return }
+		self.applicationDidBecomeActiveObserver = NotificationCenter.default.addObserver(forName: UIApplication.didBecomeActiveNotification, object: nil, queue: nil) { [weak self] _ in
+			guard let self = self else { return }
+			self.checkAppVersionDialog(for: vc)
+		}
+	}
+
+	private func removeObserver() {
+		NotificationCenter.default.removeObserver(applicationDidBecomeActiveObserver as Any)
+        applicationDidBecomeActiveObserver = nil
+	}
+
+	func createAlert(_ type: UpdateAlertType, vc: UIViewController?) -> UIAlertController? {
+		let alert = UIAlertController(title: "Akutalisierung verfügbar", message: "Es gibt eine neue Aktualisierung für die Applikation", preferredStyle: .alert)
+		alert.addAction(UIAlertAction(title: NSLocalizedString("Aktualisieren", comment: "Default action"), style: .default, handler: { _ in
+			//TODO: Add correct App Store ID
+			guard let url: URL = URL(string: "itms-apps://itunes.apple.com/app/apple-store/") else {
+				return
+			}
+				UIApplication.shared.open(url, options: [:], completionHandler: nil)
+		}))
+		switch type {
+		case .update:
+			alert.addAction(UIAlertAction(title: NSLocalizedString("Später aktualisieren", comment: "Remind me later"), style: .default, handler: { _ in
+				self.setObserver(vc: vc)
+				//Do nothing
+			}))
+		case .forceUpdate:
+			alert.message = "Um die Applikation weiter zu nutzen müssen sie eine neue Version installieren"
```

Comment:
Loc?

Reply:
Added with https://github.com/corona-warn-app/cwa-app-ios/pull/269/commits/02fc0a9450c15753f564d41a293eedafacb47084

Only german texts at the moment.